Bug: Tuple unpacking doesn't parse

## Description

Cannot unpack multiple return values from a function into separate variables in a single declaration.

## Current Behavior

```
error[E2001]: unexpected token ','
  --> test.ez:6:11
   |
6 |     temp a int, b string = getValues()
   |           ^ unexpected token encountered
```

## Expected Behavior

Should be able to declare multiple variables and unpack tuple return values:

```ez
temp a int, b string = getValues()
```

## Reproduction

```ez
do getValues() -> (int, string) {
    return 42, "hello"
}

do main() {
    temp a int, b string = getValues()  // Parse error
}
```


## Possible Fix

Update the variable declaration parser to handle comma-separated variable names with their types, then match them against the tuple return type of the function call.

--- a/pkg/ast/ast.go
+++ b/pkg/ast/ast.go
@@ -258,11 +258,13 @@ func (r *RangeExpression) TokenLiteral() string { return r.Token.Literal }
 
 // VariableDeclaration represents temp x int = 5 or const y int = 10
 // Also supports multiple assignment: temp result, err = divide(10, 0)
+// And typed tuple unpacking: temp a int, b string = getValues()
 type VariableDeclaration struct {
 	Token      Token // temp or const
 	Name       *Label
 	Names      []*Label // for multiple assignment (result, err)
 	TypeName   string
+	TypeNames  []string // for typed tuple unpacking (int, string)
 	Value      Expression
 	Mutable    bool
 	Attributes []*Attribute // @suppress(...) attributes

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -760,6 +760,58 @@ func (p *Parser) parseVarableDeclaration() *VariableDeclaration {
 		return nil
 	}
 
+	// Check for typed tuple unpacking: temp a int, b string = getValues()
+	if p.peekTokenMatches(COMMA) {
+		// Initialize TypeNames with the first type
+		stmt.TypeNames = append(stmt.TypeNames, stmt.TypeName)
+
+		// Parse additional (name type) pairs
+		for p.peekTokenMatches(COMMA) {
+			p.nextToken() // consume comma
+			p.nextToken() // move to next identifier
+
+			if !p.currentTokenMatches(IDENT) {
+				msg := fmt.Sprintf("expected identifier, got %s", p.currentToken.Type)
+				p.errors = append(p.errors, msg)
+				p.addEZError(errors.E2001, msg, p.currentToken)
+				return nil
+			}
+
+			name := p.currentToken.Literal
+			// Check for reserved names
+			if isReservedName(name) {
+				msg := fmt.Sprintf("'%s' is a reserved keyword and cannot be used as a variable name", name)
+				p.errors = append(p.errors, msg)
+				p.addEZError(errors.E2020, msg, p.currentToken)
+				return nil
+			}
+			// Check for duplicate declaration
+			if !p.declareInScope(name, p.currentToken) {
+				msg := fmt.Sprintf("'%s' is already declared in this scope", name)
+				p.errors = append(p.errors, msg)
+				p.addEZError(errors.E2023, msg, p.currentToken)
+				return nil
+			}
+			stmt.Names = append(stmt.Names, &Label{Token: p.currentToken, Value: name})
+
+			// Parse type for this variable
+			p.nextToken()
+			typeName := p.parseTypeName()
+			if typeName == "" {
+				return nil
+			}
+			stmt.TypeNames = append(stmt.TypeNames, typeName)
+		}
+
+		// Typed tuple unpacking requires initialization
+		if !p.expectPeek(ASSIGN) {
+			return nil
+		}
+		p.nextToken() // move to value
+		stmt.Value = p.parseExpression(LOWEST)
+		return stmt
+	}
+
 	// Optional initialization
 	if p.peekTokenMatches(ASSIGN) {
 		assignToken := p.peekToken // save the = token for error reporting

--- a/test/tuple_unpacking_test.ez
+++ b/test/tuple_unpacking_test.ez
@@ -1,0 +1,43 @@
+import @std
+
+do getTwo() -> (int, string) {
+    return 42, "hello"
+}
+
+do getThree() -> (int, string, bool) {
+    return 1, "test", true
+}
+
+do getNumbers() -> (int, int, int) {
+    return 10, 20, 30
+}
+
+do main() {
+    using std
+
+    println("=== Tuple Unpacking Tests ===")
+    println("")
+
+    println("Test 1: Typed tuple unpacking (two values)")
+    temp a int, b string = getTwo()
+    println("  a = ${a}")
+    println("  b = ${b}")
+
+    println("Test 2: Typed tuple unpacking (three values)")
+    temp x int, y string, z bool = getThree()
+    println("  x = ${x}")
+    println("  y = ${y}")
+    println("  z = ${z}")
+
+    println("Test 3: Untyped tuple unpacking (type inference)")
+    temp p, q = getTwo()
+    println("  p = ${p}")
+    println("  q = ${q}")
+
+    println("Test 4: Three integers")
+    temp n1 int, n2 int, n3 int = getNumbers()
+    println("  n1 = ${n1}, n2 = ${n2}, n3 = ${n3}")
+
+    println("")
+    println("=== All Tuple Unpacking Tests Passed! ===")
+}

